odoo_initializer/tests/test_integration_loader.py

```diff
@@ -0,0 +1,80 @@
+from odoo import tests
+from odoo.tests import logging
+
+from ..models import BaseLoader
+
+_logger = logging.getLogger(__name__)
+
+
+class TestLoader(tests.TransactionCase):
+    _test_model = "res.groups"
+
+    @staticmethod
+    def _get_non_updated_groups():
+        return [
+            {"id": "group_test_1", "name": "test_1", "comment": "other"},
+            {"id": "group_test_2", "name": "test_2", "comment": "filter"},
+            {"id": "group_test_3", "name": "test_3", "comment": "other"},
+        ]
+
+    @staticmethod
+    def _get_updated_groups():
+        return [
+            {"id": "group_test_1", "name": "test_1", "comment": "other"},
+            {"id": "group_test_3", "name": "test_3", "comment": "updated comment"},
+        ]
+
+    def test_no_update_rule_should_not_update_comment(self):  # Add updated groups records and test it with load_file()
+
+        # Setup
+        test_loader = BaseLoader()
+        test_loader.model_name = self._test_model
+        test_loader.field_rules = {
+            "comment": "NO_UPDATE"
+        }
+        test_loader.fields = ["id", "name", "comment"]
+        test_loader.load_file(
+            test_loader._pre_process(self._get_non_updated_groups(), None, None)
+        )
+
+        # Replay
+        processed_file = test_loader._pre_process(self._get_updated_groups(), None, None)
+
+        test_loader.load_file(
+            processed_file
+        )
+
+        # Verify
+        assert "updated comment" not in processed_file
```

Comment:
This is not quite a real integration test. We should probably test that the record is indeed created, but the comment has not been updated.
Right now this is more of a unit test, that tests that the `_pre_process` function works as expected.
1. Let's create a test Util class to fetch records in the Testing context.
2. As you've suggested @chinzou , let's add a integration test to verify that `load_file()` does its job.
3. Let's use this Util class to actually verify that the record is updated, except for the **comment**.